cqrs-user-command-service/src/test/java/dev/amir/usercommand/application/usecase/UserUseCasesTest.java

```diff
@@ -136,4 +138,40 @@ void test_DeleteUser() {
         retryFunction.execute();
         verify(userOutputPortMock).delete(eq(userId));
     }
+
+    @Test
+    void test_UpdateUser() {
+        User user = RandomObject.nextObject(User.class);
+        RetryFunctionMatcher<User> retryMatcher = new RetryFunctionMatcher<>();
+
+        User userResponse = new User();
+        userResponse.setId(new UserId());
+
+        when(retryExecutorMock.execute(argThat(retryMatcher))).thenReturn(userResponse);
+        when(userOutputPortMock.update(any(User.class))).thenReturn(user);
+
+        underTest.updateUser(user);
+
+        verify(retryExecutorMock).execute(retryFunctionCaptor.capture());
+        RetryFunction<User> retryFunction = retryFunctionCaptor.getValue();
+        retryFunction.execute();
+        verify(userOutputPortMock).update(eq(user));
+    }
+
+    @Test
+    void test_when_UserIsRemoved_ThrowsException() {
+        User user = new User();
+        user.setId(new UserId());
+        user.setStatus(UserStatus.REMOVED);
+
+        when(userOutputPortMock.isUserRemoved(any(User.class))).thenReturn(true);
+
+        UserNotFoundException exception = assertThrows(
+                UserNotFoundException.class,
+                () -> underTest.changeUserPassword(user)
+        );
+
+        assertEquals("User with id: " + user.getId() + " Not found", exception.getMessage());
+        verify(userOutputPortMock, times(1)).isUserRemoved(user);
```

Comment:
you can remove `times(1)` it is automatically defined if you don't sent that argument.
it should be:
`verify(userOutputPortMock).isUserRemoved(user);`